### PR TITLE
Nc news 4

### PR DESCRIPTION
**Pull request by AdamSaunders26**

### CODE REVIEW

File: endpoints.json [hunk 1]
```diff
@@ -25,5 +25,22 @@
...
+        "article_id": 3,
+        "title": "Eight pug gifs that remind me of mitch",
+        "topic": "mitch",
+        "author": "icellusedkars",
+        "body": "some gifs",
+        "created_at": "2020-11-03T09:12:00.000Z",
+        "votes": 0,
+        "article_img_url": "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+      }
+    }
```
Comment (Jodiech) [inline endpoints.json:44]: Nice to see you are updating your endpoints file 😄

File: db/models/article.model.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const db = require("../connection");
+
+exports.selectArticle = (article_id) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
```
Comment (Jodiech) [inline db/models/article.model.js:5]: Great to see you are making use of a parameterized query to protect against SQL injection 👍

File: db/error-handlers.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  res.status(500).send({ message: "server is broken" });
+};
+
+exports.customError = (err, req, res, next) => {
+  if (err.message) {
+    res.status(err.status).send({ message: err.message });
+  } else {
+    next(err);
+  }
+};
```
Comment (Jodiech) [inline db/error-handlers.js:12]: Nice to see you are splitting your errors by functionality here 👍

File: __tests__/app.test.js [hunk 4]
```diff
@@ -48,3 +48,38 @@ describe("GET /api/", () => {
...
+        expect(body.article).toHaveProperty("title");
+        expect(body.article).toHaveProperty("article_id");
+        expect(body.article).toHaveProperty("body");
+        expect(body.article).toHaveProperty("topic");
+        expect(body.article).toHaveProperty("created_at");
+        expect(body.article).toHaveProperty("votes");
+        expect(body.article).toHaveProperty("article_img_url");
+      });
+  });
+  test("400: should return an error message for a non-existing article_id", () => {
```
Comment (Jodiech) [inline __tests__/app.test.js]: This should be a status 404 for a non-existent id. You do need a 400 for when the id is not found 👍

File: __tests__/app.test.js [hunk 5]
```diff
@@ -48,3 +48,38 @@ describe("GET /api/", () => {
...
   });
 });
+
+describe("GET /api/articles/:article_id", () => {
+  test("200: should return the article object linked to the given id", () => {
+    return request(app)
+      .get("/api/articles/3")
+      .expect(200)
+      .then(({ body }) => {
+        expect(typeof body.article).toBe("object");
```
Comment (Jodiech) [inline __tests__/app.test.js:69]: You ideally want to write your test so it checks a particular article id, for example - article_id: 3 👍 How could you refactor this so you are using .toMatchObject?

**Review by Jodiech — COMMENTED** [review]
Hey 👋 nice work on this ticket! Please read the comments I have left and make some changes. Make some slight tweaks with your test and send another PR when you're ready 😄

**Review by hevdevs — APPROVED** [review]
Really nice Adam! 🥳 Merge away

**Pull request merged**